### ISSUE TITLE
Github responses json is not used efficiently

### ISSUE THREAD

**Issue by g4s8** [jcabi/jcabi-github#1369]
Github API json responses may contain a lot of information, e.g. issue response has all issue labels and milestones (https://api.github.com/repos/jcabi/jcabi-github/issues/1), but instead of read it from json `RtIssue` make additional requests to Github API to fetch labels or milestones. It may be a cause of https://github.com/zerocracy/farm/issues/813 bug

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by amihaiemil** [comment]
@g4s8 I think Issue has ``json()`` method or something, to get the whole json?

**Comment by amihaiemil** [comment]
@g4s8 well, apparently not, it doesn't seem to have it. Issue should definetely be able to return the whole json representation.

**Comment by g4s8** [comment]
@amihaiemil yes, but it is not used in `labels()` and `milestone()`

**Comment by amihaiemil** [comment]
@g4s8 Yes, we could improve the library somehow. But right now, simply returning ``this.json.getArray("labels")`` from method ``labels()`` turns the whole architecture upside down :D

**Comment by yegor256** [comment]
@g4s8 @amihaiemil I guess we just need to introduce a new method `Issue.Smart.listLabels()`, which will simply do `this.jsn.getArray("labels")' and iterate them. Correct?

**Comment by amihaiemil** [comment]
@yegor256 Yes we could do that to make it easier. But I looked closer and actually Issue already has the method

```java
@Override
public JsonObject json() throws IOException {
    return new RtJson(this.request).fetch();
}
```

**Comment by yegor256** [comment]
@amihaiemil yes, we have `json()`, but we don't have any code that would help us turn JSON into labels. I'm suggesting `listLabels()`

**Comment by g4s8** [comment]
@yegor256 I have same problem here: https://github.com/jcabi/jcabi-github/issues/1361
I can't return `Repo` for `/user/:user/repos` because `RtRepo` fetches repo json from network:
```java
    RtRepo(final Github github, final Request req, final Coordinates crd) {
        this.ghub = github;
        this.entry = req;
        this.coords = crd;
        this.request = this.entry.uri()
            .path("/repos")
            .path(this.coords.user())
            .path(this.coords.repo())
            .back();
    }
```
but `/user/:user/repos` has full repo json already.

**Comment by g4s8** [comment]
@yegor256 I mean that I have all user repos json after call to `/user/:user/repos`, but I can't wrap it with `RtRepo` because it fetches repo json by coordinates. Maybe we can introduce `RepoJson` which can wrap json-object and implements `Repo` interface?

**Comment by amihaiemil** [comment]
@g4s8 well, I personally think that such model wrappers (e.g. your ``RepoJson``) should not be the job of the API wrapper, the API wrapper should return the ``JsonObject`` and let the client wrap it in whatever animating model.

Because the API may change, the Json response may change, so the library shouldn't have the worry of always updating the models. That's how I design ``docker-java-api`` :)
